### PR TITLE
Add Lock-app example for mbed platform

### PR DESCRIPTION
**Pull request by lukgni**
#### Problem
Missing lock-app example for mbed platform

 #### Summary of Changes
Create initial directory structure for mbed lock-app example.
Update CMakeList.txt (copied from mbed/shell) to work with lock-app.
Add flashing and debug task for lock-app example
Update mbed_example.yaml github workflow

### CODE REVIEW

**Comment by fkjagodzinski** [comment]
Marking this as _ready for review_. We got to a point where the provisioning works for Wi-Fi (with a fixed Android `CHIPtool` app). So far I haven't tested the locking/unlocking app function; don't have a complete setup yet -- missing CHIP server.

A couple of notes:
- The impressive number of line changes (~5.5k) is mostly due to `examples/lock-app/mbed/wifi-ism43362/` dir that I copied from the _shell_ example. If we like to break the pattern of copying these files for every app I'm happy to add an update here. Suggestions are welcome. :wink: 
- This PR brings a few updates for BLE -- `src/platform/mbed/BLEManagerImpl.cpp`,
    - indication ACK hack,
    - Bluetooth pairing support.
- When testing with the `CHIPtool`, it is not mandatory to pair the device with the phone. The provisioning flow works fine without pairing.

**Comment by pan-** [comment]
I remember there was an issue with mbed import tools and symbolic links. Can you add the `wifi-ism43362` into thrid party and link it just like mbed os in the example scripts ?

File: examples/lock-app/mbed/arch.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// TODO: Remove!
+// This file is a temporary workaround until atomic integration has been solved
+
+void __sync_synchronize() { 
+    MBED_BARRIER();
+}
+
+unsigned int __atomic_fetch_add_4(volatile uint32_t *ptr, uint32_t val, int memorder) { 
+    return core_util_atomic_fetch_add_explicit_u32(ptr, val, memorder);
+}
```
Comment (ATmobica) [inline examples/lock-app/mbed/arch.c:13]: I think this file should be also moved to the common directory.  It was modified for IP integration.

**Review by ATmobica — APPROVED** [review]

**Comment by fkjagodzinski** [comment]
> (...) Can you add the `wifi-ism43362` into thrid party and link it just like mbed os in the example scripts ?

Done. Please note that  :construction: **this needs #132 to build successfully for the disco now.** :construction:

**Comment by fkjagodzinski** [comment]
Ok, with the cosmetic updates out of the way, I think this is ready for a re-review.

**Comment by fkjagodzinski** [comment]
Sorry for a force-push; had to fix the last commit. An update to `src/platform/BUILD.gn` was missing.

**Review by pan- — APPROVED** [review]

**Pull request merged**